server/src/main/java/org/candlepin/liquibase/PerOrgProductsMigrationTask.java

```diff
@@ -0,0 +1,455 @@
+                    rowbuilder.append("?,");
+                }
+                rowbuilder.deleteCharAt(rowbuilder.length() - 1).append(')');
+
+                builder = new StringBuilder(15 + 20 * cols.length + table.length() +
+                    (2 + cols.length * 3) * rows);
+
+                builder.append("INSERT INTO ").append(table).append('(');
+                for (String column : cols) {
+                    builder.append(column).append(',');
+                }
+                builder.deleteCharAt(builder.length() - 1).append(") VALUES");
+
+                for (int i = 0; i < rows; ++i) {
+                    builder.append(rowbuilder).append(',');
+                }
+                builder.deleteCharAt(builder.length() - 1);
+            }
+            else {
+                rowbuilder = new StringBuilder(20 * cols.length + table.length());
+
+                rowbuilder.append("INTO ").append(table).append('(');
+                for (String column : cols) {
+                    rowbuilder.append(column).append(',');
+                }
+                rowbuilder.deleteCharAt(rowbuilder.length() - 1).append(") VALUES(");
+                for (int i = 0; i < cols.length; ++i) {
+                    rowbuilder.append("?,");
+                }
+                rowbuilder.deleteCharAt(rowbuilder.length() - 1).append(") ");
+
+                builder = new StringBuilder(30 + (20 * cols.length + table.length()) * rows);
+                builder.append("INSERT ALL ");
+
+                for (int i = 0; i < rows; ++i) {
+                    builder.append(rowbuilder);
+                }
+
+                builder.append("SELECT 1 FROM DUAL");
+            }
+
+            return this.connection.prepareStatement(builder.toString());
+        }
+
+        return null;
+    }
+
+    /**
+     * Executes the multi-org upgrade task.
+     *
+     * @throws DatabaseException
+     *  if an error occurs while performing a database operation
+     *
+     * @throws SQLException
+     *  if an error occurs while executing an SQL statement
+     */
+    public void execute() throws DatabaseException, SQLException {
+
+        // Store the connection's auto commit setting, so we may temporarily clobber it.
+        boolean autocommit = this.connection.getAutoCommit();
+
+        try {
+            this.connection.setAutoCommit(false);
+
+            // Migrate orgs
+            ResultSet result = this.executeQuery("SELECT count(id) FROM cp_owner");
+            result.next();
+            int count = result.getInt(1);
+            result.close();
+
+            ResultSet orgids = this.executeQuery("SELECT id, account FROM cp_owner");
+            for (int index = 1; orgids.next(); ++index) {
+                String orgid = orgids.getString(1);
+                String account = orgids.getString(2);
+
+                this.logger.info(
+                    "Migrating data for org %s (%s) (%d of %d)", account, orgid, index, count
```

Comment:
We have our own wrapper which I modified a while back to accept printf-style inputs.